core/services/headtracker/head_broadcaster.go

```diff
@@ -74,6 +76,10 @@ func (hr *headBroadcaster) Close() error {
 	})
 }
 
+func (hr *headBroadcaster) SetLogger(logger *logger.Logger) {
+	hr.logger = logger
```

Comment:
Is it possible for this to race?

Reply:
Yes and... this has some very big implications IMHO, if we have to protect every access to `hr.logger` that's going to be:
1. Very error prone, you'll have to remember to do it everywhere
2. Potentially affect the overall performance of chainlink negatively because it'll add lot of locking contention and overhead.
We could probably wrap this into some kind of logContainer type wrapper to reduce verbosity but this would would only really address 1.